tests/ReferenceTest.php

```diff
@@ -71,6 +95,27 @@ public function testModelProperty()
         $this->assertSame('order', $o->table);
     }
 
+    public function testRefTraversing()
+    {
+        $user = new Model($this->db, 'user');
+        $user->addField('name');
+        $user->addField('company_id');
+
+        $user->id = 1;
+
+        $company = new Model($this->db, 'company');
+        $company->addField('name');
+        $user->hasOne('Company', [$company, 'our_field' => 'company_id', 'their_field' => 'id']);
+
+        $order = new Model($this->db, 'order');
+        $order->addField('company_id');
+        $order->addField('amount', ['default' => 20]);
+
+        $company->hasMany('Orders', [$order]);
+
+        $this->assertEquals(20, $user->ref('Company')->ref('Orders')->get('amount')); // 'amount' default value
```

Comment:
Useless test - it will always return default value

What you should do I think is:
1. move this test to ReferenceSqlTest class and revert these changes from ReferenceTest
2. instead of get(amount) use count(...->export()) or export([amount]) or sum(amount) or something like that to actually test which records will be selected
3. Also add tests for unloaded and loaded models
```
$user->unload()->ref('Company')->ref('Orders')->export(['amount']);
$user->load(1)->ref('Company')->ref('Orders')->export(['amount']);
```

Reply:
All updated